server/cmwell-dc/src/main/scala/cmwell/dc/stream/InfotonRetriever.scala

```diff
@@ -96,33 +111,25 @@ object InfotonRetriever extends LazyLogging {
               .via(
                 Framing.delimiter(endln, maximumFrameLength = maxStatementLength)
               )
-              .fold(
-                RetrieveTotals(
-                  state._1
-                    .map(im => (im.meta.path, (new ByteStringBuilder, None)))
-                    .toMap,
-                  new ByteStringBuilder
-                )
-              ) { (totals, nquad) =>
+              .fold(RetrieveTotals(HashMap(state._1 map{im => im.meta.path -> (new ByteStringBuilder, ExtraData("", None, None))}: _*), new ByteStringBuilder))
+              { (totals, nquad) =>
                 totals.unParsed ++= nquad
-                val parsed: Option[ParsedNquad] = parseNquad(remoteUri, nquad)
-                parsed.fold {
-                  RetrieveTotals(totals.parsed, totals.unParsed)
-                } { p =>
-                  totals.parsed.get(p.path) match {
+                val parsed: ParsedNquad = parseNquad(remoteUri, nquad)
```

Comment:
there are lines got from _out that are not related to any infoton. You must return Option or so to "know" about it.